### PR TITLE
Chapter 3 and beginning of chapter 4. 

### PR DESCRIPTION
**Pull request by SirVsbi**
The code so far contains:

-  all the JPA entities and flywal migration files. 
- basic GET AND POST endpoints for products and suppliers (so, no /{id} endpoints yet) 
- repo, service, dto, converter, controller for products & suppliers

### CODE REVIEW

File: src/main/java/ro/msg/learning/shop/controller/ProductController.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        return new ListResponse<>(
+                (ArrayList<ProductDto>) productConverter.convertModelToDto(productService.getAllProducts())
+        );
+    }
+
+    @PostMapping(value = "")
+    public MessageResponse addProduct(@RequestBody ProductDto productDto){
+        try{
+            System.out.println(productDto);
+            System.out.println(productConverter.dtoToModel(productDto));
```
Comment (cristinacalbaza) [inline src/main/java/ro/msg/learning/shop/controller/ProductController.java]: Remove these 2 lines.

File: src/main/java/ro/msg/learning/shop/model/ProductCategory.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "ProductCategoryId")
+    private Long productCategoryId;
+
+    private String name;
+
+    private String description;
+
+    @OneToMany(mappedBy = "productCategoryId", cascade = CascadeType.ALL, targetEntity = Product.class)
+    private List<Product> products;
```
Comment (cristinacalbaza) [inline src/main/java/ro/msg/learning/shop/model/ProductCategory.java]: As we discussed today, please remove the list from here.

File: src/main/java/ro/msg/learning/shop/controller/ProductCategoryController.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    @Autowired
+    private ProductCategoryService productCategoryService;
+
+    @Autowired
+    private ProductCategoryConverter productCategoryConverter;
+
+    @GetMapping(value = "")
+    public ProductCategoryResponse getAllProductCategories(){
+        return new ProductCategoryResponse(
+                (ArrayList<ProductCategoryDto>) productCategoryConverter.convertModelToDto(productCategoryService.getAllProductCategories())
```
Comment (cristinacalbaza) [inline src/main/java/ro/msg/learning/shop/controller/ProductCategoryController.java:28]: You should not access the Data Model from Controller. So, as it is already suggested in Training material, your service methods should receive as parameters a DTO object and return also a DTO. Please change  this for all Services in the app.

**Review by cristinacalbaza — CHANGES_REQUESTED** [review]

**Comment by SirVsbi** [comment]
Fixed the issues mentioned above. 
Moved the converter calls into the service, now the REST controller only uses DTO's. 
Implemented the rest of the REST API only for the suppliers.

**Review by cristinacalbaza — APPROVED** [review]

**Pull request merged**